src/sagemaker/session.py

```diff
@@ -4114,6 +4126,21 @@ def submit(request):
                         ModelPackageGroupName=request["ModelPackageGroupName"]
                     )
                 )
+            if "SourceUri" in request and request["SourceUri"] is not None:
+                if re.match(MODEL_PACKAGE_ARN_PATTERN, request["SourceUri"]) or re.match(
+                    MODEL_ARN_PATTERN, request["SourceUri"]
+                ):
+                    if "InferenceSpecification" in request:
+                        del request["InferenceSpecification"]
```

Comment:
should we delete inference spec or should we raise exception that both cannot be present.